### ISSUE TITLE
Add possibility to alternate between dim up and down by button hold actions

### ISSUE THREAD

**Issue by senilio** [Luligu/zigbee2mqtt-automations#2]
**Is your feature request related to a problem? Please describe.**
I would like to be able to control both on/off and dim of a lamp using a single button.

**Describe the solution you'd like**
Let's say I have a single button remote which supports short press and hold actions. I would like to be able to both toggle on/off of a lamp using short press of the button, but also be able to dim the same lamp up and down but using the hold action. The hold action should alternate between dim up and dim down. The speed of dimming should preferably be controllable via config. The dimming should occur as long as I hold the button, and stop once i release the button.

**Describe alternatives you've considered**
Currently I'm solving this by using ControllerX via Appdaemon, but would like to move this logic into z2m to get rid of the dependency to external applications.

**Additional context**
Thanks for the great extension :)

**Comment by Luligu** [comment]
Hi,
thank you for your interest.
The feature proposed is very interesting. 
I will think about how to implement it.
Did you try tuya scene button? If you bind it to a device does what you want I guess.

**Comment by senilio** [comment]
I have not tried the Tuya scene button. But if I bind for example the Ikea STYRBAR to an Ikea Lamp, the 4 buttons are pre-configured to control brightness and colors.

With my proposed feature, it would be possible to control on/off state _and_ brightness of 4 different lamps/groups of lamps using a single STYRBAR device. The big win for me would be to be able to reduce the number of remotes I need to place in each area of my home, which in turn makes it easier for other family members who want lamps to "just work" from the single remote on the wall, and they can stop trying to remember which remote goes to what :)

There is also another nice use case which I'll try to explain as short as possible. See https://www.domoticz.com/forum/viewtopic.php?p=257557&sid=249401700a7750402f3c8f201666407a#p257557

Basically solder cables to the switches on the remote PCB, which allows you to control presses with regular wall switches. This is gives great acceptance factor as no new devices needs to be introduced to the rest of the family :)

Thanks for considering the proposal!

**Comment by Luligu** [comment]
I had an idea about how to get this result. I will work on it as soon as I will have time but I guess it can work.

**Comment by wigster** [comment]
Hi. I am wondering if this is still on the cards -- it would be great to be able to have this functionality inside Z2M instead of farming it out to Home Assistant. But some kind of variable is necessary to track the direction of dimming/brightening.

**Comment by robvanoostenrijk** [comment]
I don't think it requires any code? I'm doing this with an IKEA remote using the following yaml:

Edited: Oh I just saw you want to use the same button in two modes at the same time. The automations this moment are stateless, keeping track of settable variables could make it very complex.

Leaving below for others who maybe want to use it.


```yaml
remote_bedroom_on:
  active: true
  trigger:
  - entity: Remote - Bedroom
    attribute: action
    equal: brightness_up_click
  action:
  - entity: Smart Bulb - Bedroom
    payload: turn_on

remote_bedroom_off_hold:
  active: true
  trigger:
  - entity: Remote - Bedroom
    attribute: action
    equal: brightness_down_hold
  action:
  - entity: Smart Bulb - Bedroom
    payload: { "brightness_move": -40 }

remote_bedroom_on_hold:
  active: true
  trigger:
  - entity: Remote - Bedroom
    attribute: action
    equal: brightness_up_hold
  action:
  - entity: Smart Bulb - Bedroom
    payload: { "brightness_move": 40 }

remote_bedroom_release:
  active: true
  trigger:
  - entity: Remote - Bedroom
    attribute: action
    equal: brightness_down_release
  - entity: Remote - Bedroom
    aattribute: action
    equal: brightness_up_release
  action:
  - entity: Smart Bulb - Bedroom
    payload: { "brightness_move": 0 }
```

**Comment by mrmaximas** [comment]
@robvanoostenrijk it's works if bulb support `brightness_move` feature

**Comment by wigster** [comment]
Yeah I would like to use the same button and therefore would need a state of some sort in Z2M.

**Comment by zen2** [comment]
> [@robvanoostenrijk](https://github.com/robvanoostenrijk) it's works if bulb support `brightness_move` feature

I'm using only HUE bulb devices so its support the brightness_move and brightness_step but for devices that don't support its commands it should be possible to emulate its as it's possible to get/set the brightness level.

That needs for sure some code to add the emulation and a related syntax like:

syntax:

```
action:
  - entity: the light
  - payload:
    - emulation: true
    - brightness_move: -40
```

related pseudo code:
```
  if payload.emulation is true
     if payload.brightness_move exists
        brightness_value = get(entity.brightness)+ payload.brightness_move
        payload = { " brightness": brightness_value }
```

Note that a such emulation could be implemented directly at zigbee2mqtt level too but I don't know if such feature is on track with zigbee2mqtt philosophy.
